knimin/lib/data_access.py

```diff
@@ -2735,15 +2736,22 @@ def delete_samples(self, ids):
         ------
         ValueError
             If at least one sample ID does not exist
-            If at least one sample is associated with a sample plate (in which
-            case, the sample plate has to be deleted prior to the deletion of
-            the sample)
+            If at least one sample is associated with a study
+            If at least one sample is associated with a sample plate
+            In the above two cases, the study / sample plate has to be deleted
+            prior to the deletion of the sample)
         """
         with TRN:
             for sample_id in ids:
                 if not self._sample_exists(sample_id):
                     raise ValueError('Sample ID %s does not exist.'
                                      % sample_id)
+                res = self.get_studies_associated_with_samples([sample_id])[0]
```

Comment:
In which case it is useful to delete a sample? I'm thinking that if we make the sample deletion too easy we will loose information. In any case, I think the logic here is backwards, i.e. it should be possible to delete samples associated to a study but not studies that has samples. We don't want "orphan" samples.